isis/tests/BundleSolutionInfoTests.cpp

```diff
@@ -0,0 +1,228 @@
+
+  results.setRmsImageResidualLists(rmsImageLineResiduals,
+                                   rmsImageSampleResiduals,
+                                   rmsImageResiduals);
+
+  ControlPoint *freePoint = new ControlPoint("FreePoint");
+  ControlMeasure *freeMeasure1 = new ControlMeasure;
+  freeMeasure1->SetCubeSerialNumber("Ignored");
+  freeMeasure1->SetIgnored(true);
+  freePoint->Add(freeMeasure1);
+  ControlMeasure *freeMeasure2 = new ControlMeasure;
+  freeMeasure2->SetCubeSerialNumber("NotIgnored");
+  freeMeasure2->SetIgnored(false);
+  freeMeasure2->SetCoordinate(1.0, 2.0);
+  freeMeasure2->SetResidual(-3.0, 4.0);
+  freePoint->Add(freeMeasure2);
+  SurfacePoint freeSurfacePoint(Latitude(45.0, Angle::Degrees),
+                                Longitude(120.0, Angle::Degrees),
+                                Distance(6.0, Distance::Meters));
+  freePoint->SetAdjustedSurfacePoint(freeSurfacePoint);
+  ControlPoint *fixedPoint = new ControlPoint("FixedPoint");
+  fixedPoint->SetType(ControlPoint::Fixed);
+  SurfacePoint fixedSurfacePoint(Latitude(90.0, Angle::Degrees),
+                                  Longitude(180.0, Angle::Degrees),
+                                  Distance(10.0, Distance::Meters));
+  fixedPoint->SetAdjustedSurfacePoint(fixedSurfacePoint);
+  ControlNet outNet;
+  outNet.AddPoint(freePoint);
+  outNet.AddPoint(fixedPoint);
+  BundleControlPointQsp freeBundleControlPoint(
+                                                new BundleControlPoint(settings, freePoint));
+  BundleControlPointQsp fixedBundleControlPoint(
+                                    new BundleControlPoint(settings, fixedPoint));
+  QVector<BundleControlPointQsp> bundleControlPointVector;
+  bundleControlPointVector.append(freeBundleControlPoint);
+  bundleControlPointVector.append(fixedBundleControlPoint);
+
+  BundleImage bundleImage(cube1->camera(),
+                          "Ignored",
+                          "TestImageFileName");
+  BundleObservationVector observationVector;
+  observationVector.addNew(BundleImageQsp(new BundleImage(bundleImage)),
+                           "ObservationNumber1",
+                           "Instrument1",
+                           settings);
+  
+  results.setBundleControlPoints(bundleControlPointVector);
+  results.setOutputControlNet(ControlNetQsp(new ControlNet(outNet)));
+  results.setObservations(observationVector);
+
+  BundleSolutionInfo solution(settings, networkFile, results, imageList);
+
+  solution.outputText();
+  solution.outputImagesCSV();
+  solution.outputPointsCSV();
+  solution.outputResiduals();
+
+  QFile textFile(filePrefix + "bundleout.txt");
+  QFile imagesCSV(filePrefix + "bundleout_images.csv");
+  QFile pointsCSV(filePrefix + "bundleout_points.csv");
+  QFile residualsCSV(filePrefix + "residuals.csv");
+
+  EXPECT_TRUE(textFile.exists());
+  EXPECT_TRUE(imagesCSV.exists());
+  EXPECT_TRUE(pointsCSV.exists());
+  EXPECT_TRUE(residualsCSV.exists());
+}
```

Comment:
There was one expected exception throw in the old tests and round trip serialization, do we want to add these here or make a ticket to move them over at a later date?

Reply:
I've added testing for serialization, but not the error throw as the original test's try block is empty https://github.com/USGS-Astrogeology/ISIS3/blob/dev/isis/src/control/objs/BundleSolutionInfo/unitTest.cpp#L215